benchmark/runner/mini_runners.cpp

```diff
@@ -949,6 +963,242 @@ BENCHMARK_DEFINE_F(MiniRunners, SEQ0_OutputRunners)(benchmark::State &state) {
   BenchmarkExecQuery(settings.warmup_iterations_num_ + 1, &exec_query, schema.get(), true);
 }
 
+void MiniRunners::ExecuteIndexOperation(benchmark::State *state, bool is_insert) {
+  auto key_num = state->range(0);
+  uint64_t tbl_cols = state->range(1);
+  auto num_rows = state->range(2);
+  auto type = static_cast<type::TypeId>(state->range(3));
+  auto num_index = state->range(4);
+  auto target = num_rows + 1;
+  if (settings.skip_large_rows_runs_ && num_rows > settings.warmup_rows_limit_) {
+    return;
+  }
+
+  // Skip compiled
+  if (noisepage::runner::MiniRunners::mode == execution::vm::ExecutionMode::Compiled) return;
+
+  // Create the indexes for batch-insert
+  auto cols = ConstructColumns("", type, type::TypeId::INVALID, key_num, 0);
+  auto tbl_name = ConstructTableName(type, type::TypeId::INVALID, tbl_cols, 0, num_rows, num_rows);
+  for (auto i = 0; i < num_index; i++) {
+    auto settings = GetExecutionSettings(false);
+    auto units = std::make_unique<selfdriving::PipelineOperatingUnits>();
+
+    std::stringstream query;
+    query << "CREATE INDEX idx" << i << " ON " << tbl_name << " (" << cols << ")";
+    auto equery = OptimizeSqlStatement(query.str(), std::make_unique<optimizer::TrivialCostModel>(), std::move(units),
+                                       PassthroughPlanChecker, nullptr, nullptr, &settings);
+    BenchmarkExecQuery(1, equery.first.get(), equery.second.get(), true, &empty_params, &settings);
+  }
+
+  if (!is_insert) {
```

Comment:
Why do we have to insert a tuple and the delete this specific tuple right away? I think deleting any tuple would be fine, since we're aborting the transaction anyway, right? For example, you can just delete the first tuple (`offset` = 0) from any `RawBlock` in the `DataTable`. I believe the `TupleSlot` (`RawBlock` + `offset`) is all you need for deletion.

Reply:
This makes sense. I think was mostly legacy from the TPL delete code (in case where table cardinality is not 0).